download/download.go

```diff
@@ -61,23 +67,11 @@ func New(config Config, client rest.Client, path string) *Downloader {
 }
 
 // WithCallback registers a function f to be called when download updates occur.
-func (d *Downloader) WithCallback(f func(context.Context, Update)) *Downloader {
+func (d *Downloader) WithCallback(f func(context.Context, Update) error) Interface {
```

Comment:
Is it going to be OK for us to change the public API? I was anticipating a different name for the new one that had an error response.

I guess if we assume no one else is using the downloader its maybe not a concern.